nemo/nemo/core/neural_factory.py

```diff
@@ -651,11 +651,14 @@ def deployment_export(self,
         # Custom hacks
         # We are checking type like this to avoid taking dependency on nemo_asr
         if type(module).__name__ == "JasperEncoder":
-            self.logger.warning(f"Module is JasperEncoder. We are removing"
-                                f"input and output length ports since they "
-                                f"are not needed for deployment")
-            del module._input_ports['length']
-            del module._output_ports['encoded_lengths']
+            # TK: Commenting this as this is not the way it works now.
```

Comment:
please put this back - your change will break NeMo->Jarvis export for Jasper/QuartzNet

Reply:
I cannot simply "put it back", as there are no _input_ports and _output_ports attributes of the module that you can delete from. 

My proposition is to retrieve those objects using provided methods and _then_ remove the ports you don't need.

Let's meet and discuss that.

Reply:
why remove _input_ports and _output_ports back though?

Reply:
Because those were *structures* that we don't need at all. The idea was that the methods/properties will return definitions on the fly, when called.

We have spent some significant time on those discussions and I thought we agreed on that.